.buildkite/functions.sh

```diff
@@ -1,3 +1,5 @@
+#!/bin/bash
```

Comment:
I see your reasoning, I thought about this a bit myself.

First, it's not wrong, it is just a convention used often. Then, there are a few arguments why this  is not harmful and has some benefit.

- having the shebang enables the identification of the intended shell, for example for shellcheck or, as here, codacy. Otherwise, it could be any shell dialect. However, we do use bashism from time to time, so it is worth indicating this.
- the shebang is relevant only for executable files, which both are not
- having the shebang doesn't matter. functions.sh contains only functions, executing it defines them and does not execute anything. The completion script does not do anything either, it sets up completion in the very script and then terminates, having no side effect.
- when sourcing, the shebang is a comment as any other and ignored

Especially to make files identifiable and checkable is the argument that made me accept the shebang for files intended to be sourced.

Reply:
OK I agree. I expect a shebang only if it's actually a script, but I guess it doesn't hurt. It's still uncommon, check `/etc/bash_completion.d`, there are (almost) no shebangs (depending on the packages that you have installed).

With my packager hat on, I find non-executable files with a shebang confusing, because it's not immediately clear whether the shebang is wrong or the exec bit is missing.

Reply:
Ok, accidental "unresolve" does not add the comment...

So, I can see your point, but as you mentioned yourself, there are already other packages following this scheme. I think useful (automated) checks outweigh this inconvenience.